### PR TITLE
Development jumpscale testing

### PR DESCRIPTION
**Pull request by Dina-Abd-Elrahman**
Resolves: https://github.com/threefoldtech/jumpscaleX/issues/717
## Description

Summary of the changes.

### Before asking for approval

- [ ] Make sure the changes are covered by tests.
- [ ] All the CI tasks are green.
- [ ] Ensure the branch is up to date with its base branch.
- [ ] Corresponding changes to the documentation were made.

### CODE REVIEW

File: Jumpscale/core/test/test_config_manager.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        ** create new user with certain schema, should success **
+           we will use IYO client as an example 
+
+        #. Create new github client, and save it.
+        #. Check the shema and make sure that the use is created successfully. 
+        """
+
+        client_name = self.generate_random_str()
+        self.info("Create github client {} and save it, should success".format(client_name))
+        c = j.clients.github.new(client_name, token="test_create_new_client"); c.save()
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: split it into two lines

File: Jumpscale/core/test/test_config_manager.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        ** create use with name exist before, should fail. **
+
+        #. Create user with the same name as TC198. 
+        #. Check the output, should fail. 
+        """
+
+        client_name = self.generate_random_str()
+        self.info("Create github client with {} name, should success.".format(client_name))
+        c = j.clients.github.new(client_name, token="test_create_new_client"); c.save()
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: split

File: Jumpscale/core/test/test_config_manager.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self.info("create user with the same name {}, should fail. ".format(client_name))
+        with self.assertRaises(Exception):
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: what is the Exception??

File: Jumpscale/core/test/test_config_manager.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            c=j.clients.github.new(client_name, token="test_create_new_client"); c.save()
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: split

File: Jumpscale/core/test/test_config_manager.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+        ** Test update user cleint data, should success ** 
+
+        #. Update github "test_create_client" user token. 
+        #. Check the updated token, should success. 
+        """
+
+        client_name = self.generate_random_str()
+        self.info("Update github user token")
+        c=j.clients.github.new(client_name, token="test_modify_client"); c.save()
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: split

File: Jumpscale/core/test/test_config_manager.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        c.token="test_update"; c.save()
```
Comment (0xIslamTaha) [inline Jumpscale/core/test/test_config_manager.py]: split

File: Jumpscale/core/test/test_config_manager.py (hunk 3 above)
Comment (Dina-Abd-Elrahman) [reply Jumpscale/core/test/test_config_manager.py]:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/sandbox/lib/jumpscale/Jumpscale/core/BASECLASSES/JSConfigs.py", line 93, in new
    raise RuntimeError("obj: %s already exists" % name)
RuntimeError: obj: test_delete_client_505 already exists

obj: test_delete_client_505 already exists
```

**Pull request merged**